nemo/utils/callbacks/nemo_model_checkpoint.py

```diff
@@ -454,3 +454,29 @@ def _remove_unfinished_checkpoints(checkpoint_dir: Union[Path, str]) -> None:
         # delete markers
         for marker_path in existing_marker_filepaths:
             os.remove(marker_path)
+
+    def _should_remove_checkpoint(self, trainer: "pl.Trainer", previous: str, current: str) -> bool:
```

Comment:
@stevehuang52 we want to avoid overriding PTL's protected methods in NeMo as much as possible so that NeMo doesn't break when PTL changes something in the protected methods.

Maybe we can create a PR for this in PTL itself ? Since its not a lot of modification and just addition of couple of lines in this function.

Reply:
@athitten thanks for the advice, I've created a PR to PTL https://github.com/Lightning-AI/pytorch-lightning/pull/19613, let's see if they'll approve

Reply:
Great, thank you very much @stevehuang52.

Reply:
Seems they closed the PR above. Is the resolution implemented here?

Reply:
@titu1994 Yeah they don't want to do the fix in PTL, we will do the fix in NeMo (which is this PR).